### PR TITLE
Remove old toml serialization

### PR DESCRIPTION
**Pull request by Fiono11**

### CODE REVIEW

File: config-node.toml.sample
Comment (simpago) [inline config-node.toml.sample:1]: This file should not be in git. It gets auto created during build

File: config-rpc.toml.sample
Comment (simpago) [inline config-rpc.toml.sample:1]: This file should not be in git. It gets auto created during build

File: flatc
Comment (simpago) [inline flatc:1]: This file should not be in git. It gets auto created during build

File: nano/ipc_flatbuffers_lib/flatbuffers-build/flatbuffers-config-version.cmake
Comment (simpago) [inline nano/ipc_flatbuffers_lib/flatbuffers-build/flatbuffers-config-version.cmake:1]: This file should not be in git

File: nano/core_test/toml.cpp [hunk 1]
```diff
@@ -73,7 +73,7 @@ TEST (toml, daemon_config_update_array)
 	std::filesystem::path data_path (".");
 	nano::daemon_config c{ data_path, nano::dev::network_params };
 	c.node.preconfigured_peers.push_back ("dev-peer.org");
-	c.serialize_toml (t);
+	c.serialize_toml ();
```
Comment (simpago) [inline nano/core_test/toml.cpp]: This test does nothing now, because the returned toml string is discarded. This test doesn't make much sense IMO. I would just delete it

File: nano/ipc_flatbuffers_lib/flatbuffers-build/flatbuffers.pc [hunk 2]
```diff
@@ -3,7 +3,7 @@ includedir=/usr/local/include
 
 Name: FlatBuffers
 Description: Memory Efficient Serialization Library
-Version: 1.12.0
+Version: 23.3.3
```
Comment (simpago) [inline nano/ipc_flatbuffers_lib/flatbuffers-build/flatbuffers.pc]: Why was the flatbuffers dependency updated? If this is necessary, let's put it into a new PR

File: nano/node/cli.cpp [hunk 3]
```diff
@@ -636,7 +636,7 @@ std::error_code nano::handle_node_options (boost::program_options::variables_map
 			nano::daemon_config config{ data_path, network_params };
 			// set the peering port to the default value so that it is printed in the example toml file
 			config.node.peering_port = network_params.network.default_node_port;
-			config.serialize_toml (toml);
+			config.serialize_toml ();
```
Comment (simpago) [inline nano/node/cli.cpp]: This is bug. The returned TOML string is discarded., but it should be printed to stdout

File: nano/node/daemonconfig.cpp [hunk 4]
```diff
@@ -32,13 +32,18 @@ rsnano::DaemonConfigDto to_daemon_config_dto (nano::daemon_config const & config
...
 {
 	auto dto{ to_daemon_config_dto (*this) };
-	if (rsnano::rsn_daemon_config_serialize_toml (&dto, &toml) < 0)
-		return nano::error ("could not TOML serialize daemon_config");
 
-	return toml.get_error ();
+	const size_t buffer_len = 10000;
+	std::vector<char> buffer (buffer_len);
+
+	rsnano::rsn_daemon_config_serialize_toml (&dto, buffer.data (), buffer_len);
```
Comment (simpago) [inline nano/node/daemonconfig.cpp]: Instead of a fixed buffer size it would be better return a StringDto from Rust and then use the helper function rsnano::convert_dto_to_string()

File: rust/ffi/src/config/daemon_config.rs [hunk 5]
```diff
@@ -41,17 +45,35 @@ pub unsafe extern "C" fn rsn_daemon_config_create(
 #[no_mangle]
 pub extern "C" fn rsn_daemon_config_serialize_toml(
```
Comment (simpago) [inline rust/ffi/src/config/daemon_config.rs]: As mentioned above, using StringDto would be better

File: rust/main/src/cli/commands/node/generate_config.rs [hunk 6]
```diff
@@ -25,21 +21,14 @@ pub(crate) struct GenerateConfigArgs {
...
+        let (toml_str, config_type) = if self.node {
+            let daemon_toml = DaemonToml::default();
+            (toml::to_string(&daemon_toml)?, "node")
         } else {
-            config_type = "rpc";
-            let network_constants = NetworkConstants::new(WorkThresholds::new(0, 0, 0), network);
-            let config = RpcConfig::new(&network_constants, 0);
-            config.serialize_toml(&mut toml)?
-        }
+            //let rpc_toml = RpcToml::default();
```
Comment (simpago) [inline rust/main/src/cli/commands/node/generate_config.rs:28]: Why is this commented out? Is RpcToml not ready yet?

File: rust/main/src/cli/commands/mod.rs [hunk 7]
```diff
@@ -1,4 +1,26 @@
...
+    io::{BufRead, BufReader},
+    path::PathBuf,
+};
+
 pub(crate) mod ledger;
 pub(crate) mod node;
 pub(crate) mod utils;
 pub(crate) mod wallets;
+
+fn read_toml(path: &PathBuf) -> Result<String> {
```
Comment (simpago) [inline rust/main/src/cli/commands/mod.rs]: Why is this needed. Doesn't the toml crate allow comments?

**Review by simpago — CHANGES_REQUESTED** [review]

File: rust/main/src/cli/commands/node/generate_config.rs (hunk 6 above)
Comment (Fiono11) [reply rust/main/src/cli/commands/node/generate_config.rs:28]: It is implemented in #27 .

File: nano/node/cli.cpp (hunk 3 above)
Comment (Fiono11) [reply nano/node/cli.cpp:639]: The point of returning the string was to use it to complete the config of the cli command, with the preamble and the `use_defaults` flag, so that the C++ functionality is maintained.

**Comment by simpago** [comment]
Merged with commit bdd826cc6d731e235220edcb4aad45547dbbca11
Sorry I had to squash your commits, because otherwise the accidental addition of flatc would be in  the official repository too

**Comment by simpago** [comment]
Merged with commit https://github.com/rsnano-node/rsnano-node/commit/bdd826cc6d731e235220edcb4aad45547dbbca11
Sorry I had to squash your commits, because otherwise the accidental addition of flatc would be in the official repository too
